File: tbb_roomba/src/tbb_roomba_node.cpp
#include "rclcpp/rclcpp.hpp"

class TbbRoombaNode : public rclcpp::Node {
public:
    TbbRoombaNode() : Node("tbb_roomba_node") {
        RCLCPP_INFO(this->get_logger(), "Tbb Roomba Node has been started.");
    }
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<TbbRoombaNode>());
    rclcpp::shutdown();
    return 0;
}

File: tbb_roomba/tbb_roomba/tbb_roomba_node.py
import rclpy
from rclpy.node import Node
from pycreate2 import Create2

class TbbRoombaNode(Node):
    def __init__(self):
        super().__init__('tbb_roomba_node')
        self.get_logger().info('TBB Roomba Node has been started.')
        self.roomba = Create2('/dev/ttyUSB0')

def main(args=None):
    rclpy.init(args=args)
    tbb_roomba_node = TbbRoombaNode()
    
    rclpy.spin(tbb_roomba_node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

